src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -292,16 +292,17 @@ private function parseXml(\DOMNode $node)
      *
      * @return array
      */
-    private function parseXmlAttributes(\DOMNode $node)
+    private function parseXmlAttributes(\DOMNode $node, array $context = array())
     {
         if (!$node->hasAttributes()) {
             return array();
         }
 
         $data = array();
+        $typeCastAttributes = $this->resolveXmlTypeCastAttributes($context);
 
         foreach ($node->attributes as $attr) {
-            if (!is_numeric($attr->nodeValue)) {
+            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes) {
```

Comment:
Good idea, we can even do better:

* submit the fix proposed by Nicolas into 2.7
* rebase this PR against `master` and make this feature optin (document the BC break in the CHANGELOG)

WDYT?